GAT 2023 updates for Galaxy 23.0 (#3988)
galaxyproject/training-material: topics/admin/tutorials/customization/tutorial.md

---
layout: tutorial_hands_on

title: "Customizing the look of Galaxy"
questions:
  - How do I change Galaxy's front page?
  - How do I set a custom brand text?
  - How can I configure theming?
objectives:
  - Set a brand text for your Galaxy instance
  - Add a custom welcome page to Galaxy
  - Activate and customize themes
time_estimation: "45m"
key_points:
  - Set your Galaxy instance apart from others
  - Communicate what your Galaxy instance is about
  - Offer users more customization options using themes
contributions:
  authorship:
  - ElectronicBlueberry
  editing:
  - hexylena
requirements: []
tags:
- git-gat
subtopic: maintenance
---

> Customizing your Galaxy instance makes it more recognizable at a glance, and can help communicate it's purpose to it's users.

Reviewer: ```suggestion
Customizing your Galaxy instance makes it more recognizable at a glance, and can help communicate its purpose to its users.
```

>    +    mode: "0755"
>    +
>     # Certbot
>     certbot_auto_renew_hour: "{{ 23 |random(seed=inventory_hostname)  }}"
>     certbot_auto_renew_minute: "{{ 59 |random(seed=inventory_hostname)  }}"
>    {% endraw %}
>    ```
>    {: data-commit="Add welcome url to the config"}
>
> 2. Let's then develop the associated template, add a title and some text to our welcome page:
>
>    {% raw %}
>    ```diff
>    --- /dev/null
>    +++ b/files/galaxy/welcome.html
>    @@ -0,0 +1,11 @@
>    +<!DOCTYPE html>
>    +<html lang="en">
>    +    <head>
>    +        <meta charset="utf-8">
>    +        <link href="/static/dist/base.css" rel="stylesheet" type="text/css" />
>    +    </head>
>    +    <body>
>    +        <h1>Welcome to the Galaxy Mars instance!</h1>
>    +        <p>The only Galaxy instance on mars.</p>
>    +    </body>
>    +</html>
>    diff --git a/group_vars/galaxyservers.yml b/group_vars/galaxyservers.yml
>    index 3d7fd0e..8d8bd40 100644
>    --- a/group_vars/galaxyservers.yml
>    +++ b/group_vars/galaxyservers.yml
>    @@ -86,10 +86,9 @@ galaxy_config:
>               - job-handlers
>               - workflow-schedulers
>     

> >    -galaxy_config_files:
> >    +galaxy_config_files_public:
> >       - src: files/galaxy/welcome.html
> >         dest: "{{ galaxy_mutable_config_dir }}/welcome.html"
> >    -    mode: "0755"

Reviewer: ```suggestion
>    -galaxy_config_files:
>    +galaxy_config_files_public:
>       - src: files/galaxy/welcome.html
>         dest: "{{ galaxy_mutable_config_dir }}/welcome.html"
>    -    mode: "0755"
```
I think this diff and the one in the previous step are off?

Author: Fixed, great catch.

> <hands-on-title>Configuring Themes</hands-on-title>
>
> 0. Set the location of the themes configuration in your `group_vars/galaxyservers.yml`
>
>    {% raw %}
>    ```diff
>    --- a/group_vars/galaxyservers.yml
>    +++ b/group_vars/galaxyservers.yml
>    @@ -35,6 +35,7 @@ galaxy_config:
>         # Branding
>         brand: Mars 🚀
>         logo_src: "https://training.galaxyproject.org/training-material/topics/admin/tutorials/customization/images/logo.png"
>    +    themes_config_file: "{{ galaxy_config_dir }}/themes.yml"
>         # Main Configuration
>         admin_users:
>         - admin@example.org
>    @@ -90,6 +91,10 @@ galaxy_config_files_public:
>       - src: files/galaxy/welcome.html
>         dest: "{{ galaxy_mutable_config_dir }}/welcome.html"
>     
>    +galaxy_config_files:
>    +  - src: files/galaxy/themes.yml
>    +    dest: "{{ galaxy_config.galaxy.themes_config_file }}"
>    +
>     # Certbot
>     certbot_auto_renew_hour: "{{ 23 |random(seed=inventory_hostname)  }}"
>     certbot_auto_renew_minute: "{{ 59 |random(seed=inventory_hostname)  }}"
>    {% endraw %}
>    ```
>    {: data-commit="Add brand"}
>

> > 1. Create you themes config file in `files/galaxy/themes.yml`:

Reviewer: ```suggestion
> 1. Create your themes config file in `files/galaxy/themes.yml`:
```

>    +          transparent 185px),
>    +        linear-gradient(270deg,
>    +          #3c476d 0px 110px,
>    +          #3c476d00 110px),
>    +        linear-gradient(180deg,
>    +          #FE0000 16.66%,
>    +          #FD8C00 16.66% 33.32%,
>    +          #FFE500 33.32% 49.98%,
>    +          #119F0B 49.98% 66.64%,
>    +          #0644B3 66.64% 83.3%,
>    +          #C22EDC 83.3%)
>    +      text:
>    +        color: white
>    +        hover: gold
>    +        active: white
>    +      link:
>    +        color: "#3c476d"
>    +        hover: "#323a53"
>    +        active: "#6170a6"
>    +      logo:
>    +        img: "/static/favicon.svg"
>    +        img-secondary: null
>    +
>    +smoky:
>    +  masthead:
>    +    color: "#0C0F0A"
>    +    text:
>    +      color: white
>    +      hover: "#FBFF12"
>    +      active: white
>    +    link:
>    +      color: transparent
>    +      hover: transparent
>    +      active: "#FF206E"
>    +    logo:
>    +      img: "/static/favicon.svg"
>    {% endraw %}
>    ```
>    {: data-commit="Add themes file"}
>
>    The themes file can contains themes for users to select in Galaxy.
>

> >    Each theme is identified by it's id, eg `blue`, and followed by a set of
> >    rules, which style the client. The first theme in your theme file will be
> >    used as the default theme, which users see when visiting your instance,
> >    and which users which haven't logged in will see.

Reviewer: ```suggestion
>    Each theme is identified by its id, eg `blue`, and followed by a set of
>    rules, which style the client. The first theme in your theme file will be
>    used as the default theme, which users see when visiting your instance,
>    and which users who haven't logged in will see.
```
